### ISSUE TITLE
pecl-mongodb-1.5.2 test fail: MongoDB\Driver\Exception\RuntimeException::hasErrorLabel() [tests/exception/runtimeexception-haserrorlabel-002.phpt]

### ISSUE THREAD

**Issue by hydrapolic** [mongodb/mongo-php-driver#943]
### Description
Test tests/exception/runtimeexception-haserrorlabel-002.phpt fails.

Also reproduced on other machine: https://bugs.gentoo.org/670642

### Environment
Linux Gentoo

libbson/mongo-c-driver 1.12.0
pecl-mongodb 1.5.2
php 7.2.13

### Test Script

phpize ; autoreconf ; libtoolize ; aclocal ; autoconf --force ; autoheader ; libtoolize ; ./configure ; make -j4 ; make test

### Expected and Actual Behavior

```
=====================================================================
FAILED TEST SUMMARY
---------------------------------------------------------------------
MongoDB\Driver\Exception\RuntimeException::hasErrorLabel() [tests/exception/runtimeexception-haserrorlabel-002.phpt]
```

```
================================================================================
/tmp/1/mongodb-1.5.2/tests/exception/runtimeexception-haserrorlabel-002.phpt
================================================================================
Fatal error: Uncaught Error: Call to undefined method MyException::hasErrorLabel() in /tmp/1/mongodb-1.5.2/tests/exception/runtimeexception-haserrorlabel-002.php:11
Stack trace:
#0 {main}
  thrown in /tmp/1/mongodb-1.5.2/tests/exception/runtimeexception-haserrorlabel-002.php on line 11
================================================================================
001+ Fatal error: Uncaught Error: Call to undefined method MyException::hasErrorLabel() in /tmp/1/mongodb-1.5.2/tests/exception/runtimeexception-haserrorlabel-002.php:11
002+ Stack trace:
003+ #0 {main}
001- bool(true)
002- bool(false)
003- ===DONE===
004+   thrown in /tmp/1/mongodb-1.5.2/tests/exception/runtimeexception-haserrorlabel-002.php on line 11
================================================================================
```

**Comment by jmikola** [comment]
A few things are odd here.

 * [`MongoDB\Driver\Exception\RuntimeException::hasErrorLabel()`](http://php.net/manual/en/mongodb-driver-runtimeexception.haserrorlabel.php) was introduced in 1.6.0, which has yet to be released. The method is present in the [1.6.0alpha1](https://pecl.php.net/package/mongodb/1.6.0alpha1) release, which we're using to test the development branch of [mongo-php-library](https://github.com/mongodb/mongo-php-library), since it depends on this new method.

This method was added in https://github.com/mongodb/mongo-php-driver/pull/870 for [PHPC-1222](https://jira.mongodb.org/browse/PHPC-1222). `runtimeexception-haserrorlabel-002.phpt` also does not appear to exist in this repository. The closest match for that name is `runtimeexception-haserrorlabel-001.phpt`, which was added in the aforementioned PR.

**Comment by hydrapolic** [comment]
@jmikola 
It can be found in this package: https://pecl.php.net/get/mongodb-1.5.2.tgz (from https://pecl.php.net/package/mongodb)
It's also listed in package.xml: <file md5sum="a7a85cce8563a891d44921f2669db6d3" name="tests/exception/runtimeexception-haserrorlabel-002.phpt" role="test" />

But as you noted, it does not appear in https://github.com/mongodb/mongo-php-driver/tree/1.5.2/tests/exception which is really strange, why is that so?

**Comment by jmikola** [comment]
@derickr: It looks like you released [1.5.2](https://github.com/mongodb/mongo-php-driver/releases/tag/1.5.2) on August 1. This is after https://github.com/mongodb/mongo-php-driver/pull/870 was merged on July 3. Is it possible that you had an extra test in your working directory, and that somehow got picked up by `pecl package` when prepping the tarball for the 1.5.2 release a few weeks later?

I don't think PECL would allow us to modify the tarball its hosting after the fact (even if so, that might not be kosher). I'm not sure if it's worth updating the tarball attached to the GitHub release entry, since most folks don't download that directly. Since 1.5.3 has been released, I also don't think this will affect most users installing the latest stable version.

To avoid this in the future, perhaps we can add some steps to the release process to ensure we use clean git checkouts when creating PECL packages.

**Comment by derickr** [comment]
As I have admin rights on PECL, I can replace the tarball. I have no problems with that, as it's not a code change, but just a packaging issue. But as you said, 1.5.3 has been released, so I think we should just close this problem. *Alternatively* I can unrelease the 1.5.2 release altogether, but that might break composer.json files directly (but wrongly) requiring precisely 1.5.2.

It's indeed possible that there was a rogue test case, but that would have been picked up by "make package.xml" instead of "pecl package", as the latter only processes the package.xml file.

**Comment by jmikola** [comment]
@derickr: Makes sense to leave 1.5.2 as-is since this has been addressed upstream by bumping to 1.5.3.
